src/mixins/placeholder.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+
+/**
+ * CSS to style the selection psuedo-element.
+ *
+ * @example
+ * // Styles as object usage
+ * const styles = {
+ *   ...placeholder(styles)
+ * }
+ *
+ * // styled-components usage
+ * const div = styled.input`
+ *   ...placeholder(styles)
```

Comment:
That should totally work for styled components:

```JS
const Box = styled.div`
  ${placeholder(css`styles`)}
`
```

We support arrays too, so this should work perfectly fine if just passed through, but please try it out!